test/jdk/java/lang/reflect/code/TestQuotedUnique.java

```diff
@@ -0,0 +1,35 @@
+import jdk.incubator.code.Op;
+import jdk.incubator.code.Quoted;
+import jdk.incubator.code.Reflect;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.function.IntUnaryOperator;
+import java.util.stream.IntStream;
+
+/*
+ * @test
+ * @summary For a reflectable lambda, calls to Op.ofQuotable must produce the same instance.
+ * @modules jdk.incubator.code
+ * @run junit TestQuotedUnique
+ */
+public class TestQuotedUnique {
+
+    @Test
+    public void testInSequence() {
+        int j = 8;
+        IntUnaryOperator q = (@Reflect IntUnaryOperator) i -> i * 2 + j;
+        Quoted q1 = Op.ofQuotable(q).get();
+        Quoted q2 = Op.ofQuotable(q).get();
+        Assertions.assertSame(q1, q2);
+    }
+
+    @Test
+    public void testInParallel() {
+        int j = 8;
+        IntUnaryOperator q = (@Reflect IntUnaryOperator) i -> i * 2 + j;
+        List<Quoted> quotedObjects = IntStream.range(1, 3).parallel().mapToObj(_ -> Op.ofQuotable(q).get()).toList();
```

Comment:
We need to think more carefully about this approach. I think we need N threads in proportion to the runtime threads all executing and waiting on a count down latch so all are trigger to call `Op.ofQuotable(q)` at the roughly the same time and all threads should produce the same result, returning exactly the same object or throwing UOE. Then run that process in a loop for say 100 iterations.

Reply:
So the current approach is not testing in parallel ?